kubernetes-docs/guestbook.md

```diff
@@ -0,0 +1,181 @@
+ 
+
+**Create a record for the persistent Single command [TODO]**
+
+Download all files in the PVC PV Demo from [here](https://github.com/vmware/kubernetes/tree/kube-examples/kube-examples/guestbook)
+						
+**Create all the records in one**
+
+```
+$ kubectl create -f guestbook-pvc # Folder containing all the yamls for the pvc demo 
+```
+						
+**Individual Steps**
+						
+Download the yaml files
+					
+				
+						
+Create the persistent volume records
+
+```						
+$ kubectl create -f redis-master-pv.yaml create -f redis-slave-pv.yaml
+```						
+					
+Check the persistent volume records
+
+```			
+$ kubectl get pv
+```
+
+						
+Create the PVC
+
+```
+$ kubectl create -f redis-slave-claim.yaml 
+$ kubectl create -f redis-master-claim.yaml
+```
+
+Check that the volumes are bound to the claims 
+
+``` 
+$ kubectl get pv
+ 
+NAME             CAPACITY ACCESSMODES RECLAMIPOLICY STATUS CLAIM REASON AGE
+redis-master-pv 2Gi           RWO           Retain           Bound     default/redis-master-claim     34s 
+redis-slave-pv 2Gi           RWO              Retain               Bound     default/redis-slave-claim     34s 	 		
+```
+
+Create the guestbook
+
+```
+kubectl create -f guestbook-all-in-one.yaml
+``` 
+ 
+## Guestbook Application with Dynamic Provisioning
+					
+						
+Download the yaml files for storage class based guestbook 
+ 
+Create the storage class
+
+```						
+$ kubectl create -f redis-sc.yaml
+```
+
+Create the claims that use the storage class
+
+```					
+$kubectl create -f redis-master-claim.yaml create -f redis-slave-claim.yaml
+```
+
+This should trigger storage volumes getting created on the datastore configured on the vsphere cloud provider 					
+ 
+```						
+[root@promc-2n-dhcp41:/vmfs/volumes/57f5768f-856aa050-fde c-e0db55248054/kubevols] ls *-pvc-* kubernetes-dynamic-pvc-91c9cd60-9a7c-11e6-a431-00505690d6 4d-flat.vmdk kubernetes-dynamic-pvc-a6abf43a-9a7c-11e6-a 431-00505690d64d-flat.vmdk kubernetes-dynamic-pvc-91c9cd60-9a7c-11e6-a431-00505690d6 4d.vmdk kubernetes-dynamic-pvc-a6abf43a-9a7c-11e6-a 431-00505690d64d.vmdk 
+```
+
+All the dynamic provisioned volumes will be created in the kubevols directory inside the datastore.				
+		 	 	 		
+Create the guestbook app consuming the claims 
+
+```
+$ kubectl create -f guestbook-all-in-one.yaml 
+```
+
+Verify Guestbook Application
+The following steps verify that the Pods, storage and the application are running correctly
+ 
+Wait till the pods are ready
+
+``` 
+$ kubectl get pods
+NAME                                 READY STATUS RESTARTS AGE
+frontend-88237173-77e6n 1/1         Running  0                   43m
+redis-master-1145698066-2hbbk 1/1         Running  0                   43m
+redis-slave-5845680045-3ffdg 1/1         Running  0                   43m		
+``` 
+ 
+Get the port on which the front end is accessible (kube-up does not configure an external load balancer or ingress and guestbook uses NodePort mapping)
```

Comment:
remove -> (kube-up does not configure an external load balancer or ingress and guestbook uses NodePort mapping)